File: number_guessing_game/number_guessing_game.fs
open System

let lower = 1
let upper = 100
let rnd = Random()
let n = rnd.Next(lower, upper + 1)

let rec input lower upper =
    printf $"Make a guess ({lower} - {upper}): "
    match Int32.TryParse (Console.ReadLine()) with
    | true, x when x >= lower && x <= upper -> x
    | _ -> input lower upper

let rec loop lower upper = 
    match input lower upper with
    | x when x > n ->
        printf $"{x} is too large\n"
        loop lower (x - 1)
    
    | x when x < n ->
        printf $"{x} is too small\n"
        loop (x + 1) upper
    
    | _ -> printf "You are correct\n"

loop lower upper

File: number_guessing_game/number_guessing_game.cs
using System;

var lower = 1;
var upper = 100;
var rnd = new Random();
var n = rnd.Next(lower, upper + 1);

var input = 0;
while (true)
{
    Console.Write($"Make a guess ({lower} - {upper}): ");

    if (!int.TryParse(Console.ReadLine(), out input) || input < lower || input > upper)
        continue;

    if (input > n)
    {
        Console.WriteLine($"{input} is too large");
        upper = input - 1;
    } 
    else if (input < n) 
    {
        Console.WriteLine($"{input} is too small");
        lower = input + 1;
    } 
    else 
    {
        Console.WriteLine("You are correct");
        break;
    }
}